pyincore/analyses/epnfunctionality/epnfunctionality.py

```diff
@@ -0,0 +1,176 @@
+# Copyright (c) 2019 University of Illinois and others. All rights reserved.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Mozilla Public License v2.0 which accompanies this distribution,
+# and is available at https://www.mozilla.org/en-US/MPL/2.0/
+import copy
+
+import numpy as np
+import pandas as pd
+import networkx as nx
+
+from pyincore import BaseAnalysis, NetworkDataset
+from pyincore.analyses.epnfunctionality.epnfunctionalityutil import EpnFunctionalityUtil
+
+
+class EpnFunctionality(BaseAnalysis):
+    """Computes electric power infrastructure functionality.
+    Args:
+        incore_client: Service client with authentication info
+    """
+
+    def __init__(self, incore_client):
+        super(EpnFunctionality, self).__init__(incore_client)
+
+    def run(self):
+        """Execute eletric power facility functionality analysis """
+
+        # get network dataset
+        network_dataset = NetworkDataset.from_dataset(self.get_input_dataset("epn_network"))
+        edges_epl_gdf = network_dataset.links.get_dataframe_from_shapefile()
+        nodes_epf_gdf = network_dataset.nodes.get_dataframe_from_shapefile()
+        edges_epl_gdf['weight'] = edges_epl_gdf.loc[:, 'length_km']
+        G_ep = network_dataset.get_graph_networkx()
+
+        # get epf sample
+        epf_dmg_fs = self.get_input_dataset('epf_sample_failure_state').get_dataframe_from_csv()
+        epf_sample_df = pd.DataFrame(
+            np.array([np.array(epf_dmg_fs.failure.values[i].split(',')).astype('int')
+                      for i in np.arange(epf_dmg_fs.shape[0])]),
+            index=epf_dmg_fs.guid.values)
+        # get the sample number
+        num_samples = epf_sample_df.shape[1]
+        sampcols = ['s' + samp for samp in np.arange(num_samples).astype(str)]
+
+        # add column
+        epf_sample_df.columns = sampcols
+        epf_sample_df1 = nodes_epf_gdf.loc[:, ['guid', 'nodenwid']].set_index('guid').join(epf_sample_df)
```

Comment:
I think that it will be okay but you might want to case sensitivity check in here and other lines like 53 56, 99, 102, 109 and other lines after that, but you can ignore this comment as well.